src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/ProjectPropertiesWrapper.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using NuGet.SolutionRestoreManager;
+using Microsoft.VisualStudio.ProjectSystem.PackageRestore;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore;
+
+/// <summary>
+///     Wraps a collection of <see cref="ProjectProperty"/> instances to implement the <see cref="IVsProjectProperties"/>
+///     interface for NuGet.
+/// </summary>
+internal class ProjectPropertiesWrapper : ImmutablePropertyCollection<IVsProjectProperty, ProjectProperty>, IVsProjectProperties
```

Comment:
Could these `*Wrapper` types be named `Vs*` without conflicts? I think that'd make their purpose clearer. I don't feel strongly about this though. Just an initial impression.

Reply:
I'm not happy with the `*Wrapper` names, either. I've been staring at this code for over a week and wrote these types so the names are somewhat irrelevant to me, but I'd like them to be meaningful/helpful for someone who _hasn't_ spent that much time with them.  So if you think `Vs*` would be a better naming scheme then I'll go with that.

Reply:
Done.